build-system/tasks/visual-diff/index.js

```diff
@@ -372,23 +329,39 @@ function createPercyPuppeteerController(assetGlobs) {
  *     details about the pages to snapshot.
  */
 async function generateSnapshots(percy, webpages) {
-  const numUnfilteredTests = webpages.length;
+  const numUnfilteredPages = webpages.length;
   webpages = webpages.filter(webpage => !webpage.flaky);
-  if (numUnfilteredTests != webpages.length) {
-    log('info', 'Skipping', colors.cyan(numUnfilteredTests - webpages.length),
-        'flaky tests');
+  if (numUnfilteredPages != webpages.length) {
+    log('info', 'Skipping', colors.cyan(numUnfilteredPages - webpages.length),
+        'flaky pages');
   }
   if (argv.grep) {
     webpages = webpages.filter(webpage => argv.grep.test(webpage.name));
     log('info', colors.cyan(`--grep ${argv.grep}`), 'matched',
-        colors.cyan(webpages.length), 'tests');
+        colors.cyan(webpages.length), 'pages');
+  }
+
+  // Expand all the interactive tests. Every test should have a base test with
+  // no interactions, and each test that has in interactive tests file should
+  // load those tests here.
+  for (const webpage of webpages) {
+    webpage.tests_ = {
+      '': async() => {},
```

Comment:
whats the empty func for?

Reply:
Makes sure each page has a "zero interactions" test as well (it's easier to add this empty function to the list of tests than it is to create a special case before the iterator later on)